packages/jellyfish-api-core/__tests__/category/rawtx/decodeScript.test.ts

```diff
@@ -0,0 +1,136 @@
+import { MasterNodeRegTestContainer } from '@defichain/testcontainers'
+import { Testing } from '@defichain/jellyfish-testing'
+import { wallet } from '../../../src'
+
+describe('Decodescript()', () => {
+  const container = new MasterNodeRegTestContainer()
+  const testing = Testing.create(container)
+
+  beforeAll(async () => {
+    await container.start()
+  })
+
+  afterAll(async () => {
+    await container.stop()
+  })
+
+  it('should decode empty script', async () => {
+    const address = ''
+    const decode = await testing.rpc.rawtx.decodeScript(address)
+
+    expect(decode).toStrictEqual({
+      asm: '',
+      type: 'nonstandard',
+      p2sh: '2N9hLwkSqr1cPQAPxbrGVUjxyjD11G2e1he',
+      segwit: {
+        asm: '0 e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855',
+        hex: '0020e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855',
+        reqSigs: 1,
+        type: 'witness_v0_scripthash',
+        addresses: [
+          'bcrt1quwcvgs5clswpfxhm7nyfjmaeysn6us0yvjdexn9yjkv3k7zjhp2snwgpgy'
+        ],
+        'p2sh-segwit': '2N2CmnxjBbPTHrawgG2FkTuBLcJtEzA86sF'
+      }
+    })
+  })
+
+  it('should decode scriptPubKey from BECH32 address', async () => {
+    const address = await testing.container.getNewAddress()
+    const addressInfo = await testing.rpc.wallet.getAddressInfo(address)
+    const scriptPk = addressInfo.scriptPubKey
+    const decode = await testing.rpc.rawtx.decodeScript(scriptPk)
+
+    expect(decode.asm).toStrictEqual(`${addressInfo.witness_version} ${addressInfo.witness_program}`)
+    expect(decode.reqSigs).toStrictEqual(1)
+    expect(decode.type).toStrictEqual('witness_v0_keyhash')
+    expect(decode.addresses).toStrictEqual([addressInfo.address])
```

Comment:
use object containing to do a partial match

```suggestion
expect(decode).toStrictEqual(expect.objectContaining({
  asm: `${addressInfo.witness_version} ${addressInfo.witness_program}`:
  reqSigs: 1,
  ...
}))
```